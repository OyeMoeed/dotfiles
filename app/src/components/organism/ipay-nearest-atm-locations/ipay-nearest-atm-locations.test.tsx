// ipay-nearest-atm-locations.test.tsx

import { render } from '@testing-library/react-native';
import IPayNearestAtmLocations from './ipay-nearest-atm-locations.component';

describe('<IPayNearestAtmLocations />', () => {
  it('renders markers correctly', () => {
    const { getAllByTestId } = render(<IPayNearestAtmLocations testID="test" style={{}} />);

    // Check if markers render based on the mock data
    const markers = getAllByTestId('marker-1');
    expect(markers.length).toBe(1); // Adjust this based on your actual mock data count
  });

  it('renders with default props', () => {
    const { getByTestId } = render(<IPayNearestAtmLocations testID="test" />);

    // Check if the map view container renders with default props
    const mapView = getByTestId('test-map-view-base-view');
    expect(mapView).toBeTruthy();
  });
});
