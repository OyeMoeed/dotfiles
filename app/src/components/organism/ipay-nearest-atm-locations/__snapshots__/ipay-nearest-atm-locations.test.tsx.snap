// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IPayNearestAtmLocations matches the snapshot 1`] = `
<View
  style={
    [
      {},
      [
        {
          "borderTopEndRadius": 10,
          "borderTopStartRadius": 10,
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ],
    ]
  }
  testID="test-nearest-atm-locations-map-view-base-view"
>
  MockedMapView
</View>
`;
